methods_add_cards.py

```diff
@@ -0,0 +1,265 @@
+import pandas as pd
+from typing import Tuple
+from network_wrangler import ProjectCard
+
+
+def add_cards_to_registry(
+    card_file_list: list, input_df: pd.DataFrame, config: dict, write_to_disk: bool
+) -> pd.DataFrame:
+    """
+    Returns an updated registry dataframe.
+
+    Args:
+        card_file_list: a list of project cards and their filenames
+        df: input registry DataFrame. See the format in `registry.csv`.
+        config: input configuration
+        write_cards: a boolean indicating whether project card updates should be written
+            to disk. If True, the input project cards will be overwritten.
+
+    Returns:
+        Registry DataFrame updated
+
+    """
+
+    nodes_in_use = _make_available("nodes", config)
+    links_in_use = _make_available("links", config)
+    out_df = input_df
+
+    for card, filename in card_file_list:
+        card_dict = card.__dict__
+        if card_dict["project"] not in input_df["project_added"].values:
+            node_df, node_update, updated_card_dict = _update_registry(
+                "nodes",
+                out_df,
+                card,
+                nodes_in_use,
+            )
```

Comment:
it is possible that there are no new nodes in the project card for a "New Roadway" category - when adding a new link b/w existing nodes. In that case, this would fail. I think you should run this node update part if card_dict["nodes"]  exists.